src/d3a/d3a_core/sim_results/bills.py

```diff
@@ -45,20 +32,16 @@ def update(self, area):
         self.cumulative_bills_results = {}
         self.update_cumulative_bills(area)
 
-    def _calculate_device_penalties(self, area):
-        if len(area.children) > 0:
-            return
-
-        if isinstance(area.strategy, LoadHoursStrategy):
-            return sum(
-                area.strategy.energy_requirement_Wh.get(market.time_slot, 0) / 1000.0
-                for market in _get_past_markets_from_area(area.parent, "past_markets"))
-        elif isinstance(area.strategy, PVStrategy):
-            return sum(
-                area.strategy.state.available_energy_kWh.get(market.time_slot, 0)
-                for market in _get_past_markets_from_area(area.parent, "past_markets"))
+    def _calculate_device_penalties(self, area, area_core_stats):
+        if len(area['children']) > 0 or area_core_stats == {}:
+            return 0.0
+
+        if area['type'] in ['LoadHoursStrategy', 'DefinedLoadStrategy']:
+            return area_core_stats['energy_requirement_kWh']
+        elif area['type'] in ['PVStrategy', 'PVPredefinedStrategy']:
+            return area_core_stats['available_energy_kWh']
         else:
-            return None
+            return 0.0
```

Comment:
Are you sure that this will not break the UI? If yes, ignore my comment.

Reply:
Just tested again, didn't break the UI